src/flb_io.c

```diff
@@ -174,6 +174,23 @@ int flb_io_net_connect(struct flb_connection *connection,
     return 0;
 }
 
+static void net_io_propagate_critical_error(
+                struct flb_connection *connection)
+{
+    switch (errno) {
+    case EBADF:
```

Comment:
What about ENOTTY

```
#define	ENOTTY		25	/* Not a typewriter */
```

I believe this appears as:
```
[error] [src/flb_http_client.c:1199 errno=25] Inappropriate ioctl for device
```
which is a common network error in Fluent Bit
https://github.com/aws/aws-for-fluent-bit/blob/mainline/troubleshooting/debugging.md#common-network-errors

Reply:
Uhm, I can remove `EIO` and add `ENOTTY` but I couldn't find any references to that code being a valid option for `send` or `sendto`.

Could it be by chance that this is an artifact of printing the value of errno when `flb_io_net_write` fails regardless of the reason? 

I can't really think of a scenario where that would happen but if it happens we need to find out why so if you have any insight in terms of reproduction it'd be great if you shared it with us.

Reply:
That doesn't sound good at all, do you have any leads? It'd be good if we could try to get to the bottom of it so if any of you have any information about the settings being used in those cases (I suppose it's mostly about the IO plugins) that'd be great.

Reply:
True! That would be good to figure out. Not clear if this error is coming up during the sent/sendto or during something like setting a socket to non-blocking... So it's also not clear if the error's impact would be lessened by the addition of this code. Hopefully it is though.

I see IOCTL here: https://github.com/fluent/fluent-bit/blob/450477c012307555b10dd39f07370dc55a088578/src/flb_network.c#L225-L238

Reply:
I was curious about that so I verified the code both in this branch and 1.9 but couldn't find any clues.

Reply:
> That doesn't sound good at all, do you have any leads?

Unfortunately not offhand, we've found that the log is usually benign and haven't really kept any records of it happening or tried to understand why. I've instructed the rest of the team to keep an eye out for the log in the future so we can try to provide some more info.